BuildScript.Nuke/GenerateSbom/SolutionSbomGenerator.cs

```diff
@@ -0,0 +1,204 @@
+using CycloneDX.Models;
+using CycloneDX.Utils;
+using CycloneDX.Xml;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Nuke.Common.IO;
+using Nuke.Common.ProjectModel;
+using Nuke.Common.Tooling;
+using Nuke.Common.Tools.Npm;
+using Nuke.Common.Tools.PowerShell;
+using SbomCleaner.Library;
+using Serilog;
+using Tool = Nuke.Common.Tooling.Tool;
+
+namespace Remotion.BuildScript.GenerateSbom;
+
+public class SolutionSbomGenerator: ISbomGenerator
+{
+  public SbomGeneration SbomGeneration => SbomGeneration.ForSolution;
+
+  private readonly FilterList _packageBlackList;
+  private readonly FilterList _projectBlacklist;
+
+  private readonly string _githubUsername;
+  private readonly string _githubAccessToken;
+
+  private readonly AbsolutePath _outputFile;
+
+  private readonly Solution _solution;
+  private readonly AbsolutePath _workingDirectory;
+
+  private readonly AbsolutePath? _pathToPackageJson;
+
+  public SolutionSbomGenerator (
+      Solution solution,
+      AbsolutePath workingDirectory,
+      AbsolutePath outputFile,
+      FilterList packageBlackList,
+      FilterList projectBlacklist,
+      AbsolutePath? pathToPackageJson,
+      string githubUsername,
+      string githubAccessToken)
+  {
+    _solution = solution;
+    _workingDirectory = workingDirectory;
+    _outputFile = outputFile;
+    _projectBlacklist = projectBlacklist;
+    _packageBlackList = packageBlackList;
+    _pathToPackageJson = pathToPackageJson;
+    _githubUsername = githubUsername;
+    _githubAccessToken = githubAccessToken;
+  }
+
+  public async Task Generate (Tool cycloneDxTool)
+  {
+    var uncleanedSbomFilename = "uncleaned-sbom.xml";
+
+    Directory.CreateDirectory(_workingDirectory);
+
+    cycloneDxTool.Invoke(
+        arguments:
+        $"-o . --disable-package-restore --exclude-dev --exclude-test-projects -gu {_githubUsername} -gt {_githubAccessToken} -f \"{uncleanedSbomFilename}\" {_solution.Path}",
+        workingDirectory: _workingDirectory
+    );
+
+    Log.Information("Cleaning sbom...");
+
+    var cleanedSbomFileName = "cleaned-sbom.xml";
+    var cleaner = new Cleaner();
+
+    cleaner.CleanSolutionSbom(
+        _workingDirectory / uncleanedSbomFilename,
+        _solution.Path,
+        _packageBlackList,
+        _projectBlacklist,
+        _workingDirectory / cleanedSbomFileName);
+
+    Log.Information("Created cleaned sbom '{_outputFile}'.", _workingDirectory / cleanedSbomFileName);
+
+    if (_pathToPackageJson != null)
+    {
+      var npmSbomFileName = CreateNPMSbom();
+
+      await MergeSboms(cleanedSbomFileName, npmSbomFileName);
+    }
+    else
+    {
+      Log.Information("No package json file specified, will therefore not create combined sbom.");
+
+      File.Copy(_workingDirectory / cleanedSbomFileName, _outputFile);
+
+      Log.Information($"Cleaned sbom has been copied to '{_outputFile}'.");
+    }
+  }
+
+  private async Task MergeSboms (string cleanedSbomFileName, string npmSbomFileName)
```

Comment:
I don't see any `await` calls in this method. do we need `async Task`  instead of `void`?